easybuild/easyconfigs/r/reframe/reframe-3.0-dev2.eb

```diff
@@ -17,15 +20,26 @@ sources = [
                     '%(name)s/reframe-tests-%(version)s.tar.gz'
     }
 ]
-dependencies = [('Python-bare', '3.6.10')]
+
+builddependencies = [('pip', '20.0.2', '-py3')]
+
 keepsymlinks = True
 postinstallcmds = [
     'cp %(installdir)s/config/cscs.py %(installdir)s/reframe/settings.py',
     'cp -r %(builddir)s/reframe-tests-%(version)s/checks %(installdir)s/cscs-checks/private',
-    'mv %(installdir)s/cscs-checks %(installdir)s/checks'
+    'mv %(installdir)s/cscs-checks %(installdir)s/checks',
+    'PYTHONUSERBASE=%(installdir)s/installed_by_pip pip install --user -r %(builddir)s/reframe-%(version)s/requirements.txt',
 ]
+
+modtclfooter = '''
+prepend-path PATH %(installdir)s/installed_by_pip/bin
+prepend-path PYTHONPATH %(installdir)s/installed_by_pip/lib/python3.6/site-packages
+'''
+
 sanity_check_paths = {
     'files': ['reframe.py', 'bin/reframe', 'config/cscs.py'],
     'dirs':  ['bin', 'checks', 'checks/private', 'unittests'],
 }
+
+sanity_check_commands = ['reframe -V', 'pytest --version']
```

Comment:
If you want to test `pytest`'s sanity, you can run `./test_reframe.py --version` and avoid changing the `PATH`.

Reply:
Or just do `python3 -c 'import pytest'`

Reply:
Ok I'll just do it with `python -m pytest --version`